packages/react-core/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,302 @@
+  };
+
+  const handleMousemove = (e: any) => {
+    let newPosition = e.clientX - diff - sliderRailRef.current.getBoundingClientRect().left;
+
+    const end = sliderRailRef.current.offsetWidth - thumbRef.current.offsetWidth;
+
+    const start = 0;
+
+    if (newPosition < start) {
+      newPosition = 0;
+    }
+
+    if (newPosition > end) {
+      newPosition = end;
+    }
+
+    const newPercentage = getPercentage(newPosition, end);
+
+    thumbRef.current.style.setProperty('--pf-c-slider--value', `${newPercentage}%`);
+    setValue(newPercentage);
+
+    /* If discrete, snap to closest step value */
+    if (isDiscrete && steps) {
+      const stepIndex = steps.findIndex(step => step.value >= newPercentage);
+      if (steps[stepIndex].value === newPercentage) {
+        snapValue = steps[stepIndex].value;
+      } else {
+        const midpoint = (steps[stepIndex].value + steps[stepIndex - 1].value) / 2;
+        if (midpoint > newPercentage) {
+          snapValue = steps[stepIndex - 1].value;
+        } else {
+          snapValue = steps[stepIndex].value;
+        }
+      }
+    }
+
+    // Call value change callback
+    if (onValueChange && !isDiscrete && !steps) {
+      onValueChange(newPercentage);
+    }
+  };
+
+  const callbackMouseMove = React.useCallback(handleMousemove, []);
+  const callbackMouseUp = React.useCallback(handleMouseup, []);
+
+  const handleThumbKeys = (e: React.KeyboardEvent) => {
+    const key = e.key;
+    if (key !== 'ArrowLeft' && key !== 'ArrowRight') {
+      return;
+    }
+    e.preventDefault();
+    let newValue: number = value;
+    if (isDiscrete) {
+      const stepIndex = steps.findIndex(step => step.value === value);
+      if (key === 'ArrowRight') {
+        if (stepIndex + 1 < steps.length) {
+          {
+            newValue = steps[stepIndex + 1].value;
+          }
+        }
+      } else if (key === 'ArrowLeft') {
+        if (stepIndex - 1 >= 0) {
+          newValue = steps[stepIndex - 1].value;
+        }
+      }
+    } else {
+      if (key === 'ArrowRight') {
+        newValue = value + 1 <= 100 ? value + 1 : 100;
+      } else if (key === 'ArrowLeft') {
+        newValue = value - 1 >= 0 ? value - 1 : 0;
+      }
+    }
+
+    if (newValue !== value) {
+      thumbRef.current.style.setProperty('--pf-c-slider--value', `${newValue}%`);
+      setValue(newValue);
+      if (onValueChange) {
+        onValueChange(newValue);
+      }
+    }
+  };
+
+  const inputGroup = (
+    <div className={css(styles.inputGroup)}>
+      <input
+        className={css(styles.formControl)}
+        disabled={isInputDisabled}
+        type="number"
+        value={localInputValue}
+        aria-label={inputAriaLabel}
+        onKeyDown={handleKeyPressOnInput}
+        onChange={onChangeHandler}
+        onClick={onInputFocus}
+        onFocus={onInputFocus}
+        onBlur={onBlur}
+      />
+      {inputLabel && <span className={css(styles.inputGroupText, 'pf-m-plain')}>{inputLabel}</span>}
```

Comment:
Technically an `<InputGroup>` is only needed here if this text exists, but it doesn't hurt to leave it the way it is.

Reply:
Also curious why you don't use `<InputGroupText>` for this element? Looks like we would need an enhancement for `isPlain` or something to apply `.pf-m-plain`.

I can open a separate issue for that enhancement if we don't want to do it here. It was added as part of https://github.com/patternfly/patternfly/pull/3711/

Reply:
I have updated this to use iInputGroup, `InputGroupText` and `TextInput`.  It was an oversight.  

The `InputGroupText` component  still need to be updated for the plain variation.  @mcarrano do you want me to open a new issue to make that update?  The style is still applied inline here so it still displays correctly.

Reply:
Sure. Go ahead and open a core issue @tlabaj .

Reply:
@mcarrano it is a react issue.  Core already made the update.
here is the new issue
https://github.com/patternfly/patternfly-react/issues/5367